truewiki/storage/git.py

```diff
@@ -165,6 +165,10 @@ def file_rename(self, old_filename: str, new_filename: str) -> None:
 
         super().file_rename(old_filename, new_filename)
 
+    def get_file_nonce(self, filename: str) -> str:
+        _git = git.Repo.init(self.folder)
+        return _git.head.commit.hexsha
```

Comment:
I am slightly worried about performance here. I am not sure what `git.Repo.init` does, but doing it for every nonce feels wrong. Wouldn't it be better to store the one from `prepare` in `self._git` and reuse that?

Reply:
Yes, it would. I saw you did something like that for the OutOfProcessStorage class, but it seemed like that was intentionally *not* done in this class, for what I assumed was to avoid bare calls to a git object that might be async.